devices/apa102/apa102_test.go

```diff
@@ -698,21 +702,37 @@ func BenchmarkWriteColorful(b *testing.B) {
 	_, _ = d.Write(pixels[:])
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
+		d.Intensity = uint8(i)
+		d.Temperature = uint16((3000 + i) & 0x1FFF)
 		_, _ = d.Write(pixels[:])
 	}
 }
 
-func BenchmarkDrawNRGBAColorful(b *testing.B) {
-	// Takes the fast path.
-	b.ReportAllocs()
-	img := image.NewNRGBA(image.Rect(0, 0, 150, 1))
-	for i := range img.Pix {
-		img.Pix[i] = uint8(i) + uint8(i>>8)
+func fillImage(img image.Image, f genColor) {
+	switch im := img.(type) {
```

Comment:
Oh I meant you don't need the type switch case at all. It can be rewritten as

```
func fillImage(img image.Image, f genColor) {
  for x := 0; x < im.Bounds().Dx(); x++ {
    pix := f(x)
    im.Set(x, 0, color.NRGBA{R: pix[0], G: pix[1], B: pix[2], A: 255})
  }
}
```

Reply:
Hm, I couldn't find a reasonable way to not have the type assertion because image.Image doesn't have a Set() method.  Thank for the review!

Reply:
Ah it's because you want draw.Image;
https://golang.org/pkg/image/draw/#Image